prereise/gather/demanddata/bldg_electrification/load_projection.py

```diff
@@ -0,0 +1,674 @@
+        zone_profile_refload_MWh["cool_load_mw"],
+    ) = (
+        energy_list.apply(lambda x: x[0]),
+        energy_list.apply(lambda x: x[1]),
+        energy_list.apply(lambda x: x[2]),
+        energy_list.apply(lambda x: x[3]),
+    )
+    zone_profile_refload_MWh.set_index("hour_utc", inplace=True)
+
+    # scale energy to each projection scenarios
+    elec_profile_load_mwh = {}
+    zone_profile_load_mwh = {}
+    zone_profile_load_mwh["base"] = zone_profile_refload_MWh
+    ff2hp_profile_load_mwh = {}
+    for id, scenario in new_scens.items():
+        elec_profile_load_mwh[id] = scale_energy(
+            zone_profile_refload_MWh,
+            temp_df,
+            base_scen,
+            scenario,
+            midperfhp_cop,
+            advperfhp_cop,
+        )
+        ff2hp_profile_load_mwh[id] = ff_electrify_profiles(
+            weather_years, puma_data_zone, scenario, base_scen
+        )
+        zone_profile_load_mwh[id] = pd.concat(
+            [elec_profile_load_mwh[id], ff2hp_profile_load_mwh[id]], axis=1
+        )
+
+    return zone_profile_load_mwh
+
+
+if __name__ == "__main__":
+    # Use base_year for model fitting
+    base_year = const.base_year
+
+    # Weather year to produce load profiles. If multiple years, then time series result will show for more than one year
+    weather_years = [2018, 2019]
+
+    # new heat pump load profile assumption. User can select whether to use electric profile or fossil fuel profile to estimate
+    # electrified fossil fuel consumption for heating
+    new_hp_profile = "elec"  # "elec" or "ff"
+
+    # Reading Balancing Authority and Pumas shapefiles for overlaying
+    zone_shp = read_shapefile(
+        "https://besciences.blob.core.windows.net/shapefiles/USA/balancing-authorities/ba_area/ba_area.zip"
+    )
+    pumas_shp = read_shapefile(
+        "https://besciences.blob.core.windows.net/shapefiles/USA/pumas-overlay/pumas_overlay.zip"
+    )
+
+    zone_names = [
+        "NYIS-ZONA",
+    ]
+
+    zone_name_shps = [
+        "NYISO-A",
+    ]
+
+    for i in range(len(zone_names)):
+        zone_name, zone_name_shp = zone_names[i], zone_name_shps[i]
+
+        scen_data = pd.read_csv(
+            os.path.join(
+                os.path.dirname(__file__),
+                "projection",
+                "scenario_inputs",
+                f"{zone_name}_stats.csv",
+            ),
+            index_col=0,
+        )
+
+        base_scenarios = Scenarios("base", scen_data.pop("yr2019"))
+        print(f"base scenario: year {base_year}, weather year: {weather_years}")
+
+        proj_scenarios = {}
+        for name, values in scen_data.iteritems():
+            proj_scenarios[name] = Scenarios(name, values, base_scenarios)
+            print(f"projection scenario {name}, year {proj_scenarios[name].year}")
+
+        os.makedirs(
+            os.path.join(os.path.dirname(__file__), "Profiles"),
+            exist_ok=True,
+        )
```

Comment:
Not sure we'v used this directory or not.

Reply:
Yes, the ff2hp hourly load estimations for invloved PUMAs are saved here. These files are saved because they can be used more then once when project loads for all load zones within a state.